### PR TITLE
4871 send in app notification to volunteer and supervisor on reimbursement complete checkbox selection

### LINKED ISSUE #4871: Send in-app notification to volunteer and supervisor on Reimbursement Complete checkbox selection
## What type(s) of user does this feature affect?
 - volunteers
 - supervisors
 - admins

## Description
Sign in as a `casa_admin` role. Go to https://casa-qa.herokuapp.com/reimbursements.

On checking checkbox, send an in-app notification (see Notification model) to supervisor and to volunteer saying "Volunteer Linda's request for reimbursement for 3mi for $12 on June 9, 2023 has been processed and is en route"

Screenshot 2023-06-09 at 11 09 14 AM

## QA Login Details
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by schoork**
### What github issue is this PR for, if any?
Resolves #4871 

### What changed, and why?
Added a notification that goes to volunteer and their supervisor to let them know the travel reimbursement was approved.

### How will this affect user permissions?
- Volunteer permissions: None
- Supervisor permissions: None
- Admin permissions: None

### How is this tested? (please write tests!) 💖💪
spec/system/notifications/index_spec.rb

### Screenshots please :)
Screenshot 2023-08-15 at 6 35 54 PM

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

File: app/models/learning_hour.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ def occurred_at_not_in_future
 #  id                    :bigint           not null, primary key
 #  duration_hours        :integer          not null
 #  duration_minutes      :integer          not null
+#  learning_type         :integer          default(5)
```
Comment (littleforest) [inline app/models/learning_hour.rb:33]: Should this change be here?

File: app/notifications/reimbursement_complete_notification.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  param :case_contact
+
+  def title
+    "Reimbursement Approved"
+  end
+
+  def message
+    case_contact = params[:case_contact]
+    # Questions about logic for payment rate
+    "Volunteer #{case_contact.creator.display_name}'s request for reimbursement for #{case_contact.miles_driven}mi on #{case_contact.occurred_at_display} has been processed and is en route."
```
Comment (littleforest) [inline app/notifications/reimbursement_complete_notification.rb:13]: Does this need a space between `#{case_contact.miles_driven}` and `mi`?

File: db/schema.rb [hunk 3]
```diff
@@ -382,6 +382,7 @@
 
   create_table "learning_hours", force: :cascade do |t|
     t.bigint "user_id", null: false
+    t.integer "learning_type", default: 5
```
Comment (littleforest) [inline db/schema.rb:385]: Should this be in here?

**Review by littleforest — COMMENTED** [review]
Thanks for working on this! Can we add or update a request spec that checks that the notification gets created correctly?

File: app/models/learning_hour.rb (hunk 1 above)
Comment (schoork) [reply app/models/learning_hour.rb:33]: It's from merging main into the branch. Can back it out. Looks like there won't be merge conflicts.

File: app/notifications/reimbursement_complete_notification.rb (hunk 2 above)
Comment (schoork) [reply app/notifications/reimbursement_complete_notification.rb:13]: The original ticket example had no space.

File: db/schema.rb (hunk 3 above)
Comment (schoork) [reply db/schema.rb:385]: It's from merging main into the branch. Can back it out. Looks like there won't be merge conflicts.

File: app/notifications/reimbursement_complete_notification.rb (hunk 2 above)
Comment (FireLemons) [inline app/notifications/reimbursement_complete_notification.rb:13]: I think it's supposed to include the total based on the [mileage rate](https://github.com/rubyforgood/casa/blob/main/app/models/mileage_rate.rb) When a mileage rate goes into effect all miles are reimbursed for the amount until a mileage rate with a later date goes into effect. 

In the part where the ticket says "for $12"

There could be no mileage rate.

**Review by FireLemons — CHANGES_REQUESTED** [review]

File: app/notifications/reimbursement_complete_notification.rb (hunk 2 above)
Comment (schoork) [reply app/notifications/reimbursement_complete_notification.rb]: I added some logic to set this if the casa_org has active mileage rates. Can see the main logic in CasaOrg.mileage_rate_for_given_date(DateTime)

**Review by littleforest — APPROVED** [review]
🎉

**Pull request merged**